libraries/resource_providers/image_streamer/api300/artifact_bundle.rb

```diff
@@ -0,0 +1,117 @@
+          new_name_in_use = resource_named(:ArtifactBundle).new(@item.client, name: @context.new_name).retrieve!
+          raise "#{@resource_name} name '#{@context.new_name}' is already in use." if new_name_in_use
+          Chef::Log.debug "New name #{@context.new_name} differs from current name'#{@name}' for resource."
+          @context.converge_by "Update #{@resource_name}'s name from '#{@name}' to #{@context.new_name} completed successfully" do
+            @item.update_name(@context.new_name)
+          end
+        end
+
+        def download
+          raise "#{@resource_name} #{@context.file_path} must be specified for this action" unless @context.file_path
+          raise "#{@resource_name} #{@name} does not exist in the appliance. Cannot run download action." unless @item.retrieve!
+          return Chef::Log.info("#{@resource_name} '#{@name}' file already exists in specified location.") if File.file?(@context.file_path)
+          @context.converge_by "Download #{@resource_name}'s '#{@name}.zip' to #{@context.file_path} completed successfully" do
+            @item.download(@context.file_path)
+          end
+        end
+
+        def upload
+          raise "#{@resource_name} '#{@context.file_path}' file does not exist." unless File.file?(@context.file_path)
+          return Chef::Log.info("#{@resource_name} #{@name} already exists in the appliance.") if @item.retrieve!
+          @context.converge_by "Upload of #{@resource_name}'s '#{@name}' completed successfully" do
+            @item.class.create_from_file(@item.client, @context.file_path, @name)
+          end
+        end
+
+        def extract
+          raise "#{@resource_name} #{@name} does not exist in the appliance. Cannot run extract action." unless @item.retrieve!
+          @context.converge_by "Extract #{@resource_name}'s '#{@name}'completed successfully" do
+            @item.extract
+          end
+        end
+
+        def backup
+          deployment_group = load_resource(:DeploymentGroup, @name)
+          Chef::Log.info("Starting backup of DeploymentGroup #{@name}...")
+          @context.converge_by "Backup operation of DeploymentGroup '#{@name}' completed successfully" do
+            @item.class.create_backup(@item.client, deployment_group)
+          end
+        end
+
+        def backup_from_file
+          raise 'A deployment_group field is required for this action.' unless @context.deployment_group
+          deployment_group = load_resource(:DeploymentGroup, @context.deployment_group)
+          raise "'#{@context.file_path}' file does not exist." unless File.file?(@context.file_path)
+          Chef::Log.info("Starting upload of backup file #{@context.file_path}...")
+          @context.converge_by "Upload of backup file '#{@context.file_path}' completed successfully" do
+            @item.class.create_backup_from_file!(@item.client, deployment_group, @context.file_path, @name, @context.timeout)
```

Comment:
The `@context.timeout` existence is not being checked, if it is `nil`(Not defined) it won't assume the default value.